### ISSUE TITLE
EPUB 3 Working Group Charter

### ISSUE THREAD

**Issue by iherman** [w3c/strategy#217]
New charter proposal, reviewers please take note.

# Charter Review

[Charter:](https://w3c.github.io/epub-3-wg-charter/)

What kind of charter is this? Check the relevant box / remove irrelevant branches. 

* New
 - [x] New WG 
 - [ ] New IG
 - If this is a new WG or IG charter request, link to [Advance Notice](https://lists.w3.org/Archives/Member/w3c-ac-members/2020JanMar/0036.html), and any issue discussion: [Github issues](https://github.com/w3c/epub-3-wg-charter/issues)

Horizontal Reviews (intended to be checked only by horizontal reviewers when they are satisfied with the charter)
  - [ ] Accessibility (a11y)
  - [x] Internationalization (i18n)
  - [x] Privacy
  - [x] Security

Communities suggested for outreach: 

Known or potential areas of concern?: backward compatibility with deployed EPUB 3.2; suitability of topic at W3C

Where would charter proponents like to see issues raised? (github preferred, email, ...)

- [Github issues](https://github.com/w3c/epub-3-wg-charter/issues)

Anything else we should think about as we review?

**Comment by himorin** [comment]
i18n WG would appreciate that the whole paragraph about i18n in the Scope section. No request on this draft.

**Comment by brewerj** [comment]
Thank you for the invitation to review this charter from an accessibility perspective. 
I see multiple deliverables listed in the charter, but only one, very generic, timeline. I also note that the proposed Candidate Recommendation period is long, yet W3C Process allows for completing CR in a more expedited manner, if the implementation testing work is pre-planned and managed well. It would be helpful to see more timeline detail, including specification by specification timelines; also to know if any optimization of timelines was considered. For instance, for the accessibility specification, this is timely and relevant, as people must more frequently rely upon accessible electronic publications.

**Comment by iherman** [comment]
@brewerj thanks. Would it be possible to put these comments to the charter repository? I will give my first reactions below but the proposed chairs and the publishing steering committee should be involved in the discussion...

**Comment by iherman** [comment]
My first reaction: I have to disagree on the CR period length. Getting all the testing right, as well as getting all the implementations lined up, is probably the greatest challenge in my view. A year CR will probably be needed.

Also: the various specs are very much intertwined, I don't think that separating them in terms of timeline, or publish them separately wouldn't really make sense. (The accessibility document may be an outlier in this respect, though.)

But the proposed chairs are in a much better position to give a more informed judgement on this.

**Comment by samuelweiler** [comment]
> The following features are out of scope, and will not be addressed by this Working Group group.
> 
> DRM.

Given the above, why are the specs carrying around blobs for DRM?

https://www.w3.org/publishing/epub32/epub-ocf.html#sec-container-metainf-rights.xml

**Comment by iherman** [comment]
> > The following features are out of scope, and will not be addressed by this Working Group group.
> > DRM.
> 
> Given the above, why are the specs carrying around blobs for DRM?
> 
> https://www.w3.org/publishing/epub32/epub-ocf.html#sec-container-metainf-rights.xml

This has been around in EPUB 3 (and, I presume, in EPUB 2). The strict backward compatibility requirement in the charter makes removing this entry impossible.

That being said, the container you refer to is just that: a container. What EPUB 3 does not define, and this WG will not define, is what the content of that container is, ie, the full-blown DRM specification. There are DRM regimes out there, some essentially open source, some fully opaque, but we are, and will be, absolutely silent about those.

**Comment by iherman** [comment]
@brewerj, as a pure coincidence, see https://github.com/w3c/epub-3-wg-charter/issues/15#issuecomment-619209600 as well as https://github.com/w3c/epub-3-wg-charter/issues/27 : both show that the current timeline may actually be considered as too short...

**Comment by samuelweiler** [comment]
> That being said, the container you refer to is just that: a container. What EPUB 3 does not define, and this WG will not define, is what the content of that container is, ie, the full-blown DRM specification. There are DRM regimes out there, some essentially open source, some fully opaque, but we are, and will be, absolutely silent about those.

If we want an interoperable epub format, let's define it.  By including these opaque blobs, the WG would be defining a partial spec - it would be kicking the can down the road - and it might or might not get interoperability depending on where the can lands.  

> This has been around in EPUB 3 (and, I presume, in EPUB 2). The strict backward compatibility requirement in the charter makes removing this entry impossible.

In that case, I propose that removing the container take precedence over the backward compatibility.

Indeed, removing the container will help with interoperability.  Looking at it another way, one could ask "backward compatible to what?".  Lacking full specs for the blobs, there's nothing to test against.  How about we instead say "backward compatible for documents that don't include blobs".

**Comment by iherman** [comment]
@samuelweiler,

>> That being said, the container you refer to is just that: a container. What EPUB 3 does not define, and this WG will not define, is what the content of that container is, ie, the full-blown DRM specification. There are DRM regimes out there, some essentially open source, some fully opaque, but we are, and will be, absolutely silent about those.
>
> If we want an interoperable epub format, let's define it. 

Intellectually speaking, you are probably right. But we know that other W3C members would veto any attempt to introduce such a work item into the charter. Some due to philosophical reasons, some (e.g., Apple, Adobe, or Rakuten) because that would be seen as harming their current deployed business.

Besides, that ship has largely sailed: there is a generic, portable ebook DRM system being defined and corrently driven through ISO: [LCP](https://www.edrlab.org/readium-lcp/). I do not think that W3C should get into this.

> In that case, I propose that removing the container take precedence over the backward compatibility.

is simply not an option. Please look a whole series of issues on the charter development repository: https://github.com/w3c/epub-3-wg-charter/issues/23, https://github.com/w3c/epub-3-wg-charter/issues/8, https://github.com/w3c/epub-3-wg-charter/issues/20, https://github.com/w3c/epub-3-wg-charter/issues/19. TL;DR: mainly the Publishing Industry is absolutely adamant on requiring that we maintain a very strict backward compatibility (this discussion is currently spearheaded by the Japanese industry, but others have the same reaction). If we even give the impression of going against that, they will formally oppose the creation of that working group.

**Comment by samuelweiler** [comment]
> We can't touch the container without risk to backwards compatibility,

I would like a more detailed analysis of this.  In particular, how would prohibiting the DRM blob in new versions of the epub format present a problem?  (Readers, if they support the old format, presumably understand the blob in that version of the epub format, right?)

**Comment by iherman** [comment]
"prohibiting the DRM blob" would mean that an EPUB 3.2 publication containing the DRM blob would be non-conformant. As a consequence, epubcheck would flag the usage of that thing as an error. If an EPUB publication's check reports an error then a distributor like Apple or Google will not accept it in their catalogue. 

The current charter text includes these sentences:

> The primary goal of EPUB 3.X is to remain compatible with existing content. Any existing valid EPUB 3.2 should remain valid under EPUB 3.X, unless it relies on features discovered to have serious issues (such as a security bug).

In the success criteria:

> 2. Existing content is supported. All conforming EPUB 3.2 publications should remain conformant to EPUB 3.X.
> ...
> 4. Changes that affect the global ecosystem (including authors, publishers, distributors, retailers, reading systems, and tool makers) are minimized, and made only after careful consideration of how they would affect all constituencies.

We have spent the large portion of the discussions in the past few weeks to come up with these statements to ensure that some of the participants and potential contributors would be on board. What you propose would go against that, and there is no chance this would be acceptable for those members.

**Comment by samuelweiler** [comment]
> unless it relies on features discovered to have serious issues ...

DRM is a serous issue.

So what's the point of versioning this as anything other than 3.2?  

(What I'd love here, for this discussion, is a matrix explaining the compatibility desires.  What do you want to have happen when a five-year-old epub reader gets a 3.2.1 doc?  or a 3.2 doc?  (presumably that is allowed to break, but gently?)  What do you want to have happen when a 3.2.1 reader gets a 3.2 doc?)

**Comment by wareid** [comment]
> (What I'd love here, for this discussion, is a matrix explaining the compatibility desires. What do you want to have happen when a five-year-old epub reader gets a 3.2.1 doc? or a 3.2 doc? (presumably that is allowed to break, but gently?) What do you want to have happen when a 3.2.1 reader gets a 3.2 doc?)

EPUB 3.2 is completely compatible with a reading system that processes EPUB 3.0.1. EPUB 3.x (which is what we're calling it in the charter for now), will have to meet the same requirements. 

That is the requirement for EPUB specifications. If we revise the EPUB specification in any way, it needs to remain compatible with current in-market systems. The only potential change we're proposing is to essentially make the EPUB "tent" of support larger to include properly incubated features. We know that some of those features might not be compatible with older reading systems, but as with any new feature, we'll have to seek implementation support. The proposed charter makes it clear we won't be removing any implemented features though.

**Comment by dauwhe** [comment]
> (What I'd love here, for this discussion, is a matrix explaining the compatibility desires. What do you want to have happen when a five-year-old epub reader gets a 3.2.1 doc? or a 3.2 doc? (presumably that is allowed to break, but gently?) What do you want to have happen when a 3.2.1 reader gets a 3.2 doc?)

Reading systems MUST try to process [older versions](https://www.w3.org/publishing/epub32/epub-spec.html#confreq-rs-backward-epub), and SHOULD try to process [newer versions](https://www.w3.org/publishing/epub32/epub-spec.html#confreq-rs-forward-epubn). 

> It MUST attempt to process any given rendition of an EPUB Publication whose Package Document version attribute is less than "3.0".

> It SHOULD attempt to process any given Rendition of an EPUB Publication whose Package Document version attribute is greater than "3.0".

Most of what is in EPUB is HTML and CSS, where user-agent handling of unsupported features is well-defined. The rest is either metadata (which usually doesn't affect how user agents display content) or structural information (which is of course critical, and has remained essentially unchanged for twenty years). 

Over the long term EPUB is evolving very slowly, largely by removing restrictions that are not "webby." We used to have a profile listing exactly what CSS and HTML features were allowed in EPUB. Now we largely say "just use /tr/html5" and the CSS snapshot. We're slowly deprecating features invented by EPUB in favor of web standards (for example, we had lots of prefixed CSS properties to support vertical writing mode). We occasionally allow new media types, like WOFF2 fonts. But CSS automatically provides fallbacks if the user agent doesn't support WOFF2.

**Comment by samuelweiler** [comment]
> EPUB 3.2 is completely compatible with a reading system that processes EPUB 3.0.1. EPUB 3.x (which is what we're calling it in the charter for now), will have to meet the same requirements.

Thank you for that clarification!!!

A naive reading suggests that if a 3.X doc - which didn't have a DRM blob (by fiat) - would still be readable by a 3.0.1 reader.  Taking out a blob shouldn't break that compatibility, right?

> That is the requirement for EPUB specifications. If we revise the EPUB specification in any way, it needs to remain compatible with current in-market systems. 

It sounds like this change falls under that umbrella.

**Comment by iherman** [comment]
@samuelweiler your argument seems to rely on the assumption that we should take out that blob from future versions of EPUB 3. That falls under

> Changes that affect the global ecosystem (including authors, publishers, distributors, retailers, reading systems, and tool makers) are minimized, and made only after careful consideration of how they would affect all constituencies.

that I quoted above. Such a change would seriously affect the global ecosystem, and I would not think this is an acceptable option for the industry.

**Comment by dauwhe** [comment]
I'm not sure what we could even do about "the blob". `rights.xml` is only a reserved filename, nothing more. We do say that the optional `encryption.xml`, if used, should follow an existing W3C standard (the open source Readium LCP uses `encryption.xml`). We allow essentially any file in the `META-INF` directory; and implementations such as Apple's take advantage of this fact for purposes other than DRM. Removing reserved filenames would not change the validity of any existing or future EPUB, but would also not discourage existing DRM systems.

Note that DRM serves an important purpose: It enables ebook lending by libraries around the world.

**Comment by iherman** [comment]
Charter is out for [AC review](https://lists.w3.org/Archives/Member/w3c-ac-members/2020AprJun/0060.html). See also the [Review form](https://www.w3.org/2002/09/wbs/33280/epub3wg/).

**Comment by iherman** [comment]
WG [has been announced](https://lists.w3.org/Archives/Member/w3c-ac-members/2020JulSep/0023.html). Closing.
